source/common/common/utility.h

```diff
@@ -92,4 +92,10 @@ class StringUtil {
    */
   static bool startsWith(const std::string& source, const std::string& start,
                          bool case_sensitive = true);
+
+  /**
+   * @return original @param input string if it's not empty or @param default_value otherwise.
+   */
+  static const std::string& valueOrDefault(const std::string& input,
+                                           const std::string& default_value = "-");
```

Comment:
nit: having a default param of "-" for a common utility function is a little strange, I would just have no default param.